packages/vats/src/core/startWalletFactory.js

```diff
@@ -223,32 +225,60 @@ export const startWalletFactory = async (
     privateArgs: {
       storageNode: walletStorageNode,
       walletBridgeManager,
-      walletReviver: E(ppFacets.creatorFacet).getWalletReviver(),
+      walletReviver: E(E.get(ppFacetsP).creatorFacet).getWalletReviver(),
     },
     label: 'walletFactory',
   });
   walletFactoryStartResult.resolve(wfFacets);
   instanceProduce.walletFactory.resolve(wfFacets.instance);
 
-  await Promise.all([
-    E(ppFacets.creatorFacet).addRevivableAddresses(oldAddresses),
-    E(ppFacets.creatorFacet).setReferences({
-      bankManager,
-      namesByAddressAdmin,
-      walletFactory: wfFacets.creatorFacet,
-    }),
-    publishRevivableWalletState(oldAddresses, marshaller, walletStorageNode),
-  ]);
-  const bridgeHandler = await E(ppFacets.creatorFacet).makeHandler();
+  await publishRevivableWalletState(
+    oldAddresses,
+    marshaller,
+    walletStorageNode,
+  );
 
-  await Promise.all([
-    E(provisionWalletBridgeManager).initHandler(bridgeHandler),
-    E(E.get(econCharterKit).creatorFacet).addInstance(
-      ppFacets.instance,
-      ppFacets.governorCreatorFacet,
-      'provisionPool',
-    ),
-  ]);
+  // The following initialization can only be done after the provisionPool
+  // settles, and that requires the economicCommittee
+  const initAfterProvisionPool = async () => {
+    trace('initAfterProvisionPool awaiting provisionPoolFacets');
+    const ppFacets = await ppFacetsP;
+    trace('initAfterProvisionPool settled provisionPoolFacets');
+
+    provisionPoolStartResult.resolve(ppFacets);
+    instanceProduce.provisionPool.resolve(ppFacets.instance);
+
+    await Promise.all([
+      E(ppFacets.creatorFacet).addRevivableAddresses(oldAddresses),
```

Comment:
I would dearly like to get rid of the whole reviving business.
someday...

Reply:
... over a rainbow?